packages/glimmer-runtime/lib/dom/helper.ts

```diff
@@ -104,39 +167,7 @@ class DOMHelper {
   }
 
   insertHTMLBefore(_parent: Element, nextSibling: Node, html: string): Bounds {
-    // TypeScript vendored an old version of the DOM spec where `insertAdjacentHTML`
-    // only exists on `HTMLElement` but not on `Element`. We actually work with the
-    // newer version of the DOM API here (and monkey-patch this method in `./compat`
-    // when we detect older browsers). This is a hack to work around this limitation.
-    let parent = _parent as HTMLElement;
-
-    let prev = nextSibling ? nextSibling.previousSibling : parent.lastChild;
-    let last;
-
-    if (html === null || html === '') {
-      return new ConcreteBounds(parent, null, null);
-    }
-
-    if (nextSibling === null) {
-      parent.insertAdjacentHTML('beforeEnd', html);
-      last = parent.lastChild;
-    } else if (nextSibling instanceof HTMLElement) {
-      nextSibling.insertAdjacentHTML('beforeBegin', html);
-      last = nextSibling.previousSibling;
-    } else {
-      // Non-element nodes do not support insertAdjacentHTML, so add an
-      // element and call it on that element. Then remove the element.
-      //
-      // This also protects Edge, IE and Firefox w/o the inspector open
-      // from merging adjacent text nodes. See ./compat/text-node-merging-fix.ts
-      parent.insertBefore(this.uselessElement, nextSibling);
-      this.uselessElement.insertAdjacentHTML('beforeBegin', html);
-      last = this.uselessElement.previousSibling;
-      parent.removeChild(this.uselessElement);
-    }
-
-    let first = prev ? prev.nextSibling : parent.firstChild;
-    return new ConcreteBounds(parent, first, last);
+    return insertHTMLBefore(this.uselessElement, _parent, nextSibling, html);
```

Comment:
At one time this was `abstract` however it seems like there needs to be a concrete implementation, because we need to apply HTML fixes for tree construction that delegate back to `super` in certain cases.